src/main/java/com/klarna/hiverunner/builder/HiveShellBuilder.java

```diff
@@ -61,7 +61,8 @@ public void addResource(String targetFile, String data) throws IOException {
         resources.add(new HiveResource(targetFile, data));
     }
 
-    public void setScriptsUnderTest(List<Path> scripts, Charset charset) {
+    public List<Script> setScriptsUnderTest(List<Path> scripts, Charset charset) {
```

Comment:
Let's change this method signature back and introduce a new method called `public List<Script> fromScriptPaths(List<Path> scriptPaths, Charset charset)` which does everything this method does except for adding the returned Scripts to `scriptsUnderTest`. Then this method can call the new method and add the results to `scriptsUnderTest`. Downstream we can then call this new method to convert from paths to scripts without having any side effects.